src/utils/common-utils.ts

```diff
@@ -11,49 +11,184 @@ export enum ResourceType {
-        case InitiatorType.INPUT:
-            return ResourceType.IMAGE;
-        case InitiatorType.IFRAME:
-        case InitiatorType.FRAME:
-            return ResourceType.DOCUMENT;
-        case InitiatorType.SCRIPT:
-            return ResourceType.SCRIPT;
-        case InitiatorType.CSS:
-        case InitiatorType.LINK:
-            return ResourceType.STYLESHEET;
-        default:
-            return ResourceType.OTHER;
+export const performanceKey = (details: HasLatency) =>
+    [details.startTime, details.duration].join('#');
+
+const extensions = [
+    {
+        name: ResourceType.STYLESHEET,
+        list: ['css', 'less']
+    },
+    {
+        name: ResourceType.DOCUMENT,
+        list: ['htm', 'html', 'ts', 'doc', 'docx', 'pdf', 'xls', 'xlsx']
+    },
+    {
+        name: ResourceType.SCRIPT,
+        list: ['js']
+    },
+    {
+        name: ResourceType.IMAGE,
+        list: [
+            'ai',
+            'bmp',
+            'gif',
+            'ico',
+            'jpeg',
+            'jpg',
+            'png',
+            'ps',
+            'psd',
+            'svg',
+            'tif',
+            'tiff'
+        ]
+    },
+    {
+        name: ResourceType.FONT,
+        list: ['fnt', 'fon', 'otf', 'ttf', 'woff']
+    }
+];
+
+export const shuffle = (a: any[]) => {
+    for (let i = a.length - 1; i > 0; i--) {
+        const j = Math.floor(Math.random() * (i + 1));
+        const v = a[i];
+        a[i] = a[j];
+        a[j] = v;
     }
 };
 
+export const getResourceFileType = (
+    url: string,
+    initiatorType?: string
+): ResourceType => {
+    let ext = ResourceType.OTHER;
+    if (url) {
+        const filename = url.substring(url.lastIndexOf('/') + 1);
+        const extension = filename
+            .substring(filename.lastIndexOf('.') + 1)
+            .split(/[?#]/)[0];
+
+        extensions.forEach((type) => {
+            if (type.list.indexOf(extension) > -1) {
+                ext = type.name;
+            }
+        });
+    }
+
+    /**
+     * Resource name sometimes does not have the correct file extension names due to redirects.
+     * In these cases, they are mislablled as "other". In these cases, we can infer the correct
+     * fileType from the initiator.
+     */
+    if (initiatorType && ext === ResourceType.OTHER) {
+        switch (initiatorType) {
+            case InitiatorType.IMAGE:
+            case InitiatorType.IMG:
+            case InitiatorType.INPUT:
+                ext = ResourceType.IMAGE;
+                break;
+            case InitiatorType.IFRAME:
+            case InitiatorType.FRAME:
+                ext = ResourceType.DOCUMENT;
+                break;
+            case InitiatorType.SCRIPT:
+                ext = ResourceType.SCRIPT;
+                break;
+            case InitiatorType.CSS:
+                ext = ResourceType.STYLESHEET;
+                break;
+        }
+    }
+    return ext;
+};
+
+/* Helpers */
+export const httpStatusText = {
```

Comment:
Nit: this is an unused object that we can remove in a separate PR before the next release.